### PR TITLE
Docs: Added Upsolver to vendor list

### PR DESCRIPTION
**Pull request by jasonf20**
This PR simply adds Upsolver to the vendor list docs page.

### CODE REVIEW

**Review by nastra — APPROVED** [review]

**Review by bitsondatadev — COMMENTED** [review]
I would suggest you remove a lot of the redundant phrasing that is implied by this being a service built on Iceberg.

File: site/docs/vendors.md [hunk 1]
```diff
@@ -71,3 +71,7 @@ Starburst is a commercial offering for the [Trino query engine](https://trino.io
...
+
+### [Upsolver](https://upsolver.com)
+
+[Upsolver](https://upsolver.com) is a simple to use, high volume streaming data ingestion and table management solution for Apache Iceberg. With Upsolver, users can easily ingest batch and streaming data from files, streams and databases (CDC) into [Iceberg tables](https://docs.upsolver.com/reference/sql-commands/iceberg-tables/upsolver-managed-tables). In addition, Upsolver connects to your existing lakes cataloged by Tabular or AWS Glue Data Catalog, and [analyzes the health](https://docs.upsolver.com/how-to-guides/apache-iceberg/optimize-your-iceberg-tables) of your tables. Use Upsolver to continuously optimize tables by compacting small files, sorting and compressing, repartitioning, and cleaning up dangling files and expired manifests. Upsolver is available from the [Upsolver Cloud](https://www.upsolver.com/sqlake-signup-wp?utm_page=iceberg.org) or can be deployed in your AWS VPC.
```
[the remark this answers is not in the document]
Comment (jasonf20) [reply site/docs/vendors.md:77]: Sure, submitted and update with this suggestion. Thanks!

[the remark this answers is not in the document]
Comment (bitsondatadev) [reply site/docs/vendors.md:77]: Hey @jasonf20 im on my phone so maybe it's not refreshing but I don't see the update. I'll be back at my computer shortly to check on this.

[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply site/docs/vendors.md:77]: @bitsondatadev is right. @jasonf20 didn't update the PR (forgot to push may be). 

I do have a question about this statement, "Upsolver connects to your existing lakes cataloged by Tabular or AWS Glue Data Catalog" or the suggested statement with "REST and Hive catalogs", 

Does this mean, Upsolver doesn't work with the other native Iceberg catalogs like JDBC, Nessie, HMS?
If we are using the generic catalog API, then it should work with all catalogs and we can simplify the statement as "Upsolver connects to your existing Iceberg catalog"

[the remark this answers is not in the document]
Comment (jasonf20) [reply site/docs/vendors.md]: Sorry, I pushed to the wrong repository, the fix should be there now.

[the remark this answers is not in the document]
Comment (jasonf20) [reply site/docs/vendors.md]: Currently Upsolver doesn't support JDBC / Nessie Iceberg catalogs, I agree that the generic phrasing doesn't necessarily mean we support all catalogs though.

**Review by bitsondatadev — APPROVED** [review]
LGTM! Thanks @jasonf20

File: site/docs/vendors.md [hunk 2]
```diff
@@ -71,3 +71,7 @@ Starburst is a commercial offering for the [Trino query engine](https://trino.io
...
+
+### [Upsolver](https://upsolver.com)
+
+[Upsolver](https://upsolver.com) is a streaming data ingestion and table management solution for Apache Iceberg. With Upsolver, users can easily ingest batch and streaming data from files, streams and databases (CDC) into [Iceberg tables](https://docs.upsolver.com/reference/sql-commands/iceberg-tables/upsolver-managed-tables). In addition, Upsolver connects to your existing REST and Hive catalogs, and [analyzes the health](https://docs.upsolver.com/how-to-guides/apache-iceberg/optimize-your-iceberg-tables) of your tables. Use Upsolver to continuously optimize tables by compacting small files, sorting and compressing, repartitioning, and cleaning up dangling files and expired manifests. Upsolver is available from the [Upsolver Cloud](https://www.upsolver.com/sqlake-signup-wp?utm_page=iceberg.org) or can be deployed in your AWS VPC.
```
[the remark this answers is not in the document]
Comment (jasonf20) [reply site/docs/vendors.md]: changed

**Review by nastra — APPROVED** [review]

**Pull request merged**